include/dav1d/dav1d.rs

```diff
@@ -1,3 +1,10 @@
+// TODO(kkysen)
+// In C, these are incomplete types.
+// This can be mimicked in Rust with type alias impl trait,
+// but that is still unstable, so the types are just `pub` for now.
+pub use crate::src::internal::Dav1dContext;
+pub use crate::src::r#ref::Dav1dRef;
```

Comment:
I'm not sure I understand why we need TAIT for this? If these types are part of the public API but the fields shouldn't be exposed, shouldn't we just make the fields private (or `pub(crate)` if needed internally)? C doesn't have privacy modifiers so exposing the type as an incomplete type is the only way to hide fields, but in Rust we have proper privacy control.

Reply:
Hmm, I was previously thinking that we still expose other things like the size of the type if we just make all the fields private, but that might be the case for TAIT, too, now that I think about it.  Making all the fields `pub(crate)` is probably enough.

Reply:
I've now changed it in bac5f540c4cde6e1c5ecdf84c940dac5e30c81d3 to make all the fields `pub(crate)` instead.